File: 3rd assignment/palindrom.cpp
#include<iostream>
using namespace std;
class Palindrom
{
    public:void display()
    {
        int n, num,digit,rev;
        cout<<"Enter a positive no:";
        cin>>num;

        n=num;
        do
        {
            digit = num%10;
            rev =(rev*10)+digit;
            num=num/10;
        }
        while (num!=0);
        cout<<"The reverse of a number is:"<<rev<<"\n";

        if (n==rev)
            cout<<"the number is palindrom";
            else
            cout<<"the number is not a palindrom";
    }
};
int main()
{
    Palindrom Numbers ;
    Numbers.display();
}

File: JAVA/Assignment3/Palindrom.java
import java.util.Scanner;
class Palindrom
{
	public static void main(String[] args)
	{
	    int n,num,rem,rev=0;
		Scanner sc=new Scanner(System.in);
		System.out.println("enter the no");
		num=sc.nextInt();
		n=num;
		do
		{
		rem=num%10;
		rev=(rev*10)+rem;
		num=num/10;
		}
		while(num!=0);
		{
	    System.out.println("reverse no="+rev);
		}
        if (n==rev)
            System.out.println("the number is palindrom");
            else
            System.out.println("the number is not a palindrom");
	}
}
